web_src/css/base.css

```diff
@@ -344,6 +344,12 @@ a.muted,
   color: inherit;
 }
 
+button.muted,
+.muted-buttons button {
```

Comment:
`button.muted` doesn't mean the same as `a.muted`?

`.muted-buttons` is never used, and I guess it wouldn't help much in the future, remove it?

Reply:
`.muted-buttons` was in use once where go code generated the buttons, but I think it's obsolete already and can be removed.

Reply:
In history, such design does cause problems.

For example:

```html
<div class="muted-buttons">
    <button ...>
    <div class="another-pannel">
        <button do-not-want-muted>
    </div>
</div>
```

The real case is #23888 ..... Fomantic UI used `:not()` selector to make `<button do-not-want-muted>` happy, but it's really a mess.

Reply:
Yes, I'm a strong supporter of eliminating HTML tags in translations.

Reply:
Do we have a definition of "muted"? What is a muted `a` or a muted `button`?

Reply:
> Because color in UI "screams" for attention

Agreed. And so does displaying something as a (raw) button.
So, here's a new attempt to define `muted`:
Remove the attention an element generates through its default appearance.
That's what I've tried to state above but have struggled with.

---

> The `button.muted` isn't general enough at the moment.

(In which way are you even arguing here? Should the selector only be `muted` without the `button`, or should it be more specific?)
Either way, it is **supposed** to be like that.
Anything more general probably won't work as there are still differences between what it should do for `links` and `button`.
Anything more specific is not a good idea for something that is supposed to be a helper class to make it easy to remove the default appearance of a button.

> in the future, we need to add hover styles for these markdown toolbar elements

Why should this not be possible?
It isn't `!important`, so the pseudo-class selector will have a higher specificity and overwrite the `muted` class…

Reply:
> > in the future, we need to add hover styles for these markdown toolbar elements
> 
> Why should this not be possible? It isn't `!important`, so the pseudo-class selector will have a higher specificity and overwrite the `muted` class…

What's your plan of doing so? Usually toolbar buttons should have visual feedbacks (eg: GitHub's toolbar buttons have hover styles)

If you do:

```
// in base.css
button.muted { ... }

// in editor-markdown.css
markdown-toolbar button:hover {}

// then
<markdown-toolbar>
    <button class="muted">...</button>
<markdown-toolbar>
```

How about just do this instead:

```
// in editor-markdown.css
markdown-toolbar button {}
markdown-toolbar button:hover {}

// then
<markdown-toolbar>
    <button>...</button>
<markdown-toolbar>
```


I believe the second one is clearer and it's easier to be fine-tuned for various cases.

Reply:
TBH, I also need such buttons, and it already exists in Fomantic library: `ui button tertiary`.  The oauth key generation / repo home topic manage buttons are all `ui button tertiary`.

So, I think we had better have a unique button definition for these cases, instead of `muted`, I prefer to describe a kind of element by its function and purpose, instead of using antonym words for them (eg: what styles/attributes/behaviors are "muted"?).

Reply:
We do need a class for buttons that look like links and which does only the color addition. I suggest `btn-link` that does not depend on fomantic.

```
.btn-link:hover {
  color: var(--color-primary);
}
```

Reply:
Let's move to thread below, this one is getting too long.